compiler/rustc_middle/src/mir/interpret/queries.rs

```diff
@@ -115,15 +115,10 @@ impl<'tcx> TyCtxt<'tcx> {
                     // @lcnr believes that successfully evaluating even though there are
                     // used generic parameters is a bug of evaluation, so checking for it
                     // here does feel somewhat sensible.
-                    if !self.features().generic_const_exprs() && ct.args.has_non_region_param() {
-                        let def_kind = self.def_kind(instance.def_id());
-                        assert!(
-                            matches!(
-                                def_kind,
-                                DefKind::InlineConst | DefKind::AnonConst | DefKind::AssocConst
-                            ),
-                            "{cid:?} is {def_kind:?}",
-                        );
+                    if !self.features().generic_const_exprs() 
+                        && ct.args.has_non_region_param()
+                        && self.def_kind(instance.def_id()) == DefKind::AnonConst
```

Comment:
I think there should be a comment explaining why we check the def_kind here.

Reply:
doh.... yeah, why would I put it only in the test :laughing:

Reply:
updated. let me know if the comment doesn't make sense to you and it needs elaborating somewhere :)

Reply:
I barely know what this entire function does, let alone what the FCW is for, so I'm afraid I am mostly lost here, but this way at least there are some breadcrumbs to follow for someone who stumbles upon this curious extra check in the future. :)